investments/serializers/subscription_create.py

```diff
@@ -1,51 +1,74 @@
 from decimal import Decimal
-
 from django.db import transaction
-from django.utils.translation import gettext_lazy as _
+from django.db.models import Sum
 from rest_framework import serializers
 
-from investors.models import Investor
+from investments.models import Subscription
 from projects.models import Project
-from ..models import Subscription
-from ..services.investment_share_service import calculate_investment_share
-from ..services.subscription_validation_service import validate_subscription_business_rules
-
+from investors.models import Investor
 
 class SubscriptionCreateSerializer(serializers.ModelSerializer):
-    investor = serializers.IntegerField()
-    project = serializers.IntegerField()
-    amount = serializers.DecimalField(max_digits=18, decimal_places=2, min_value=Decimal("0.01"))
-
+    """
+    Serializer for creating a new investment subscription.
+    Handles validation for investment amount and project status.
+    Uses atomic updates to prevent race conditions.
+    """
     class Meta:
         model = Subscription
-        fields = ['id', 'investor', 'project', 'amount', 'investment_share', 'created_at']
-        read_only_fields = ['investment_share', 'created_at']
-
+        fields = ['project', 'amount']
+        read_only_fields = ['investor']
+ 
     def validate(self, data):
+        project = data.get('project')
+        amount = data.get('amount')
+        request = self.context.get('request')
+        user = request.user
+
         try:
-            investor = Investor.objects.get(pk=data['investor'])
+            investor = user.investor
         except Investor.DoesNotExist:
-            raise serializers.ValidationError({"investor": _("Investor does not exist.")})
+            raise serializers.ValidationError({"investor": "The requesting user is not an investor."})
 
-        try:
-            project = Project.objects.get(pk=data['project'])
-        except Project.DoesNotExist:
-            raise serializers.ValidationError({"project": _("Project does not exist.")})
+        if project.startup and investor.user == project.startup.user:
+            raise serializers.ValidationError({"non_field_errors": "You cannot invest in your own project."})
+
+        current_funding = project.subscriptions.aggregate(total=Sum("amount"))["total"] or Decimal("0.00")
 
-        amount = data['amount']
-        exclude_amount = Decimal('0.00')
+        if current_funding >= project.funding_goal:
+            raise serializers.ValidationError({"project": "This project is already fully funded."})
+
+        if amount is None:
+            raise serializers.ValidationError({"amount": "This field is required."})
```

Comment:
There is no explicit check for whether the project exists. If project is not a valid Project instance, this may raise an AttributeError later. Add a check to ensure project is not None and is of the correct type.